pkg/cloud/libvirt/actuators/machine/actuator.go

```diff
@@ -408,7 +408,15 @@ func (a *Actuator) applyMachineStatus(
 	}
 
 	machineCopy := machine.DeepCopy()
-	machineCopy.Status.ProviderStatus = rawStatus
+
+	// Check if machine status has been modified.
+	machineStatus, err := ProviderStatusFromMachine(a.codec, machine)
+	if err != nil {
+		return false, err
+	}
+	if status != nil || !equality.Semantic.DeepEqual(machineStatus, status) {
```

Comment:
I'm unsure about the `if status != nil || ` part of the check.
It currently is: "If status is not `nil`, then set `machineCopy.Status.ProviderStatus` to `rawStatus`" or "if status is `nil`, then compare status against `machineStatus`"